src/modules/QtQuick/ColorAnimation.js

```diff
@@ -0,0 +1,132 @@
+const EMPTY_ANIMATION_COLOR = new QmlWeb.QColor();
+
+QmlWeb.registerQmlType({
+  module: "QtQuick",
+  name: "ColorAnimation",
+  versions: /.*/,
+  baseClass: "PropertyAnimation",
+  properties: {
+    from: "color",
+    to: "color",
+  }
+}, class {
+  constructor(meta) {
+    QmlWeb.callSuper(this, meta);
+
+    this.$at = 0;
+    this.$loop = 0;
+
+    QmlWeb.engine.$addTicker((...args) => this.$ticker(...args));
+    this.runningChanged.connect(this, this.$onRunningChanged);
+  }
+  $redoActions() {
+    this.$actions = [];
+    for (let i = 0; i < this.$targets.length; i++) {
+      for (const j in this.$props) {
+        const target = this.$targets[i];
+        const property = this.$props[j];
+        const from_color = this.from;
+        const to_color = this.to;
+        // Animations always take time in SequentialAnimation
+        // , regardless of the value from and to
+        const action = {
+          // from, to
+          target,
+          property,
+          from_color,
+          to_color
+        };
+        this.$actions.push(action);
+      }
+    }
+    this.$startLoop();
+  }
+  $startLoop() {
+    for (const i in this.$actions) {
+      const action = this.$actions[i];
+      const from_color = action.from_color ||
+        action.target[action.property] ||
+        EMPTY_ANIMATION_COLOR;
+      const to_color = action.to_color ||
+        action.target[action.property] ||
+        EMPTY_ANIMATION_COLOR;
+
+      action.from = {
+        r: from_color.r,
+        g: from_color.g,
+        b: from_color.b,
+        a: from_color.a
+      };
+      action.to = {
+        r: to_color.r,
+        g: to_color.g,
+        b: to_color.b,
+        a: to_color.a
+      };
+    }
+    this.$at = 0;
+  }
+  $ticker(now, elapsed) {
+    if (!this.running && this.$loop !== -1 || this.paused) {
+      // $loop === -1 is a marker to just finish this run
+      return;
+    }
+    if (this.$at === 0 && this.$loop === 0 && !this.$actions.length) {
+      this.$redoActions();
+    }
+    this.$at += elapsed / this.duration;
+    if (this.$at >= 1) {
+      this.complete();
+      return;
+    }
+    for (const i in this.$actions) {
+      const action = this.$actions[i];
+      const property = action.target.$properties[action.property];
+      const progress = this.easing.$valueForProgress(this.$at);
+
+      const {
+        from,
+        to
+      } = action;
+
+      const cur_color = new QmlWeb.QColor(
+        from.r + (to.r - from.r) * progress,
+        from.g + (to.g - from.g) * progress,
+        from.b + (to.b - from.b) * progress,
+        from.a + (to.a - from.a) * progress
+      );
+
+      // Notification changes
+      property.set(cur_color, QmlWeb.QMLProperty.ReasonAnimation);
```

Comment:
_Ignore this comment, just future optimization thoughts_.

This one, on the contrary, is called on each tick.
I wonder if this could be optimized…
Perhaps that could be done via creating a special QColor instance for each animation only at the start, and modifiying that with a special rgba-setter (that I could add, if needed).

Ignore this comment for now, I will think about optimizing that later myself (or just ignore that if future benchmarks will show that everything is ok here).

Note: QColor, unlike other basic types, is pretty optimized already — it doesn't create getters/setters for its properties in the runtime, those are all defined in the class itself.